app/controllers/api/v0/classified_listings_controller.rb

```diff
@@ -0,0 +1,79 @@
+module Api
+  module V0
+    class ClassifiedListingsController < ApiController
+      include ClassifiedListingsToolkit
+      respond_to :json
+
+      before_action :set_classified_listing, only: %i[show update]
+      before_action :authenticate!, only: %i[create update]
+
+      # skip CSRF checks for create and update
+      skip_before_action :verify_authenticity_token, only: %i[create update]
+
+      def index
+        @classified_listings = ClassifiedListing.published.
+          order("bumped_at DESC").
+          includes(:user, :organization, :taggings)
+
+        @classified_listings = if params[:category].present?
+                                 @classified_listings.where(category: params[:category])
+                               else
+                                 @classified_listings.limit(12)
+                               end
+
+        set_surrogate_key_header "classified-listings-#{params[:category]}"
+      end
+
+      def show
+        # rendering with json builder
+      end
+
+      def create
+        super
```

Comment:
is this because of the toolkit?

Reply:
Hello @rhymes. I added in the controller create and update methods even though they are defined in the toolkit. And added super. It is because of the before_action and rspec-rubocop that requires that action defined in before_action be explicitly defined.